freckle-http/library/Freckle/App/Http.hs

```diff
@@ -75,18 +75,14 @@ module Freckle.App.Http
   , StdMethod (..)
   ) where
 
-import Freckle.App.Prelude
+import Relude
```

Comment:
I've been putting it off because I don't know.

Just based on these refactors, migration to `Relude` looks like:

- Change `T.pack` to `toText`, a few other substitutions along these lines.
- Change `BSL.ByteString` to `LByteString`
- Remove strict `ByteString` type import
- Remove `decodeUtf8` et al imports
- Remove some MTL imports
- Hide `get` which Relude reexports from MTL
- Import `UTCTime`
- Replace `readMay` with `readMaybe`
- Replace `headMay` with `viaNonEmpty head`
- Import `for` (which I can't stand that Relude refuses to add).

Mostly better, a few things worse, it's sort of a wash.

I think I'm leaning toward

- The smaller and more open-sourcy the package, the more I'm inclined to use a popular prelude to maximize its legibility outside the org. And switching prelude in packages of these size is easy.
- For application code that's big and internal, I like having a custom prelude (though I wish it were defined _based on_ relude) to give us more control over the bulk of stuff we work with day-to-day. And because we already use a custom prelude and switching wholesale seems like a lot of pain for not enough gain.
 
Does that make sense?

Reply:
> The smaller and more open-sourcy the package, the more I'm inclined to use a popular prelude

Our [current guide](https://github.com/freckle/guides/blob/main/haskell-best-practices.md#prelude) for this is to not use an alternative prelude at all for our (non-`freckle-app`) OSS. We don't want using our OSS to bring a prelude choice. Therefore, no such packages should be using `Freckle.App.Prelude` now, and none of them should use `Relude` even if that's our team decision for non-OSS.

> For application code that's big and internal, I like having a custom prelude (though I wish it were defined based on relude)

`freckle-app` is kind of in the middle. It's not "big and internal", but it's also not "open-sourcy" (for the purposes of such decisions). It's only public to ease our own distribution practicalities, and perhaps add some design pressure to keeping things business-domain-agnostic. We don't apply any of the usual OSS rules to it (versioning, older-dependency-support, low dependencies, etc).

> Does that make sense?

Because its category is vague, I can't quite tell if you saying "I like having a custom prelude (though I wish it were defined based on relude)" applies to `freckle-app` or not.

Maybe you don't consider it "big and internal", so you're _not_ doing that in this PR, and that's why you're using `Relude` instead of extracting `freckle-app-prelude`.

On the other hand, maybe you do consider it "not open-sourcy", and so you _are_ doing that in this PR, but you're doing so by creating package-local preludes that are based on `Relude`.

Does my distinction make sense?

Reply:
I am not sure if by `freckle-app` you mean the package or the git repository. I think each of the different packages being split apart here incurs a separate question as to whether its fate is to be a "true OSS" thing (`base` `Prelude`) or an "us" thing (organization's prelude choice).

Reply:
I've taken out `relude` entirely since I think these breakout packages can aspire to usual OSS rules.

Reply:
> I think each of the different packages being split apart here incurs a separate question as to whether its fate is to be a "true OSS" thing

I was referring to `freckle-app` as it was, and all the `freckle-app-*`s you are breaking out _at the point you break them out and continue to name them that way_ (e.g "now"). 

If, for example, `freckle-app-http` seems to be generally-useful enough that it should be "true OSS", we would rename it to drop the `freckle-app-` prefix, and it would then have to abide the "no custom prelude rules". Until then, it could but doesn't have to. It's still just "what was `freckle-app`", IMO.

> I think these breakout packages can aspire to usual OSS rules

I'm on board with that.

Reply:
Yeah, so that's the

> not much consideration has been given to the package names

Probably time to give more consideration. `env` and `kafka` are already out there, but here on the table are `http`, `otel`, `memcached`. I don't know, is `freckle-` any worse than `-extra` or whatever generic disambiguator people normally throw on to a package name?

The HTTP one is probably the best candidate for renaming because it's a pretty serious piece of work and probably something that other people would really benefit from using. I'm at a loss for any meaningful naming though. `http-client-extravaganza`

Reply:
> `freckle-` any worse than `-extra` or whatever

It is, IMO, because we've established it pretty well as meaning "a thing that's not useful for others".

> The HTTP one is probably the best candidate

Agreed.

> I'm at a loss for any meaningful naming though

Agreed.

Even `Network.HTTP.Simple` is just part of `http-conduit`, which depends on `http-client`.

Maybe... `http-kit`? Like it's a (tool)kit for pagination, caching, etc?

We could always go cute, like we did with Blammo. Fine something pun- or pop-reference-based?

Reply:
> http-kit

Somehow already in use, though `http-toolkit` is free.

> We could always go cute

`httparty`?

Reply:
Favoring getting this done over naming it perfectly.